src/timestamp.cpp

```diff
@@ -0,0 +1,53 @@
+#include "g3log/time.hpp"
+#include "g3log/timestamp.hpp"
+#include <sstream>
+
+
+namespace g3 {
+
+
+	Timestamp::Timestamp(std::string& defaultFormat) :
```

Comment:
This timestamp struct is overcomplicating it and basically re-invents something that already exists.

What reasons are there to not use the already existing `timeval struct`?  As far as I understand it already exists on Windows and Linux. 

Ref: 
1) http://souptonuts.sourceforge.net/code/gettimeofday.c.html
2) https://msdn.microsoft.com/en-us/library/windows/desktop/ms740560(v=vs.85).aspx

With this functionality you can get the `std::time_t ` and microsecond precision which is what you want, right?

If the `timeval struct` works great then I  see no reason for the `TimeStamp` class and the generator.   In that case I think you can simplify this whole pull request to be much less changes. 

A) use the timeval struct and remove the std::time_t part from the `g3::logmessage`
B) use the formatting exactly as before (revert your changes). Just when the std::string formatted string is needed then the `timeval struct` can be converted to a `std::time_t` and then formatted as before. 
C) use the` timeval struct`'s microsecond precision instead of the g3 "microsecond" counter that you didn't like. 
D) remove the microsecond counter.